app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditScreenTest.kt

```diff
@@ -2878,6 +2911,104 @@ class VaultAddEditScreenTest : BaseComposeTest() {
         composeTestRule.assertNoDialogExists()
     }
 
+    @Test
+    fun `Fido2UserVerification event should prompt for user verification`() {
+        every {
+            biometricsManager.promptUserVerification(
+                onSuccess = any(),
+                onCancel = any(),
+                onLockOut = any(),
+                onError = any(),
+                onNotSupported = any(),
+            )
+        } just runs
+        mutableEventFlow.tryEmit(VaultAddEditEvent.Fido2UserVerification(true))
+        verify {
+            biometricsManager.promptUserVerification(any(), any(), any(), any(), any())
+        }
+    }
+
+    @Test
+    fun `Fido2UserVerification onSuccess should send UserVerificationSuccess action`() {
+        every {
+            biometricsManager.promptUserVerification(
+                onSuccess = captureLambda(),
+                onCancel = any(),
+                onLockOut = any(),
+                onError = any(),
+                onNotSupported = any(),
+            )
+        } answers {
+            lambda<() -> Unit>().invoke()
+        }
+        mutableEventFlow.tryEmit(VaultAddEditEvent.Fido2UserVerification(isRequired = true))
+        verify { viewModel.trySendAction(VaultAddEditAction.Common.UserVerificationSuccess) }
+    }
+
+    @Test
+    fun `Fido2UserVerification onCancel should send UserVerificationCancelled action`() {
+        every {
+            biometricsManager.promptUserVerification(
+                onSuccess = any(),
+                onCancel = captureLambda(),
+                onLockOut = any(),
+                onError = any(),
+                onNotSupported = any(),
+            )
+        } answers {
+            lambda<() -> Unit>().invoke()
```

Comment:
How does this know which lambda it is invoking?

Reply:
It knows because I'm only capturing one lambda using `captureLambda`. It's a convenience method for working with a single lambda argument. If I wanted to capture multiple lambdas I would use `slot<T>()` like...
```
val onNotSupportedSlot = slot<() -> Unit>()
val onSuccessSlot = slot<() -> Unit>()
every { 
  manager.register(
    onSuccess = capture(onSuccessSlot),
    onNotSupported = capture(onNotSupportedSlot),
  )
} answers {
  onNotSupportedSlot.captured.invoke()
}
```

Reply:
really cool!